theano/tensor/elemwise.py

```diff
@@ -2034,9 +2028,8 @@ def grad(self, inp, grads):
         prod_in, = inp
         gz, = grads
 
-        out = self(*inp)
 
-        if (out.dtype in theano.tensor.discrete_dtypes or
+        if (out[0].dtype in theano.tensor.discrete_dtypes or
```

Comment:
travis fail as there is 2 empty lines here. pep8 request a max of 1 empty line inside function/method. Just remove one empty line.

After that, we can merge.